Add support for nested key translations

thaqebon opened the pull request:
Hi,

I added support for JSON fields, allowing nested keys to be translatable.
You can now specify key paths separated by ```->```, which is the same notation used in the framework for to set values for JSON keys

```php
$translatable = ['setting->footer->copyright'];
```

All methods for setting and retrieving translations will work with nested keys, but I've excluded the ability to pass nested keys to mutation methods.
Additionally, I modified the loading of translatable attribute casts to occur at initialization time rather than loading them on demand.

thaqebon commented:
### Enable mutators for nested fields

Now mutators can be used.
```php
protected function settingsFooterCopyright(): Attribute
{
    return Attribute::make(
        get: fn (string $value) => ucfirst($value),
        set: fn (string $value) => strtolower($value),
    );
}
```
```php
public function setSettingsFooterCopyrighAttribute($value)
{
    $this->attributes['settings->footer->copyright'] = strtolower($value);
}
```
```php
public function getSettingsFooterCopyrighAttribute($value)
{
    return ucfirst($value);
}
```

freekmurze commented:
Thank you! Could you also update the readme with some friendly human readable examples of the new functionality.

I see that there are also tests failing. Is this is caused by you using new framework features, feel free to drop support for older Laravel / PHP versions.

thaqebon commented:
Thanks for the feedback! I've added friendly examples to the README and refactored the test file, so there’s no need to drop support for older Laravel versions.

freekmurze commented:
Thanks!
